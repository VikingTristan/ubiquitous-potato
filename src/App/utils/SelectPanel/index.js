import React, { Component } from "react";
import { NavLink, withRouter } from "react-router-dom";
import PropTypes from "prop-types";

import SearchBox from "../SearchBox/index";

class NavGroup extends Component {
    constructor (props) {
        super(props);
        this.state = { isActive: props.location.pathname.includes(props.route.path) };
    }

    toggleActive () {
        this.setState({ isActive: !this.state.isActive });
    }

    shouldComponentUpdate (nextProps, nextState) {
        return this.state !== nextState;
    }

    render () {
        const { title, routes } = this.props.route;

        return (
            <div className={`nav-group${this.state.isActive ? " active" : ""}`}>
                <div className="nav-heading">
                    <i className="material-icons" onClick={() => this.toggleActive()}>keyboard_arrow_right</i>
                    <span onClick={() => this.toggleActive()}>{title}</span>
                </div>
                <ul>
                    {routes.map((childRoute, i) => (
                        <li key={`nav_link_${i}`}>
                            <NavLink activeClassName="active" to={childRoute.path}>{childRoute.title}</NavLink>
                        </li>
                    ))}
                </ul>
            </div>
        );
    }
}

const SelectPanel = ({ routes }) => (
    <div className="doc-sidebar">
        <SearchBox routes={routes} />
        <nav className="documentation-nav">
            {routes.map((route, i) => {
                const NavGroupWithRouter = withRouter(NavGroup);

                return <NavGroupWithRouter key={`nav_group_${i}`} route={route} />;
            })}
        </nav>
    </div>
);

SelectPanel.propTypes = {
    routes: PropTypes.arrayOf(PropTypes.shape({
        title: PropTypes.string.isRequired,
        path: PropTypes.string.isRequired,
        redirect: PropTypes.string.isRequired,
        routes: PropTypes.arrayOf(PropTypes.shape({
            title: PropTypes.string.isRequired,
            path: PropTypes.string.isRequired,
            componentPath: PropTypes.string.isRequired
        })).isRequired
    })).isRequired
};

export default withRouter(SelectPanel);

/* For testing */
export { SearchBox };
