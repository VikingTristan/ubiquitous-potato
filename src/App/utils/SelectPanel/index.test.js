import React from "react";
import { shallow } from "enzyme";

import SelectPanel, { SearchBox } from "./index";

describe("Documentation: SelectPanel", () => {
    it("is defined", () => {
        expect(SelectPanel).toBeDefined();
    });

    // it("renders", () => {
    //     const location = { pathname: "test" };
    //     const wrapper = shallow(
    //         <SelectPanel location={location} />
    //     );

    //     Expect(wrapper).toMatchSnapshot();
    // });

    describe("SearchBox", () => {
        it("is defined", () => {
            expect(SearchBox).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<SearchBox routes={[]} />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});

