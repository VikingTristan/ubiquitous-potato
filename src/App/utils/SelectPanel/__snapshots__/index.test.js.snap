// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Documentation: SelectPanel SearchBox renders 1`] = `
<form
  className="doc-search"
>
  <input
    autoComplete="off"
    className="form-control"
    name="designguide-search"
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Search..."
    type="text"
  />
</form>
`;
